### DISCUSSION TITLE
_ctx.route is not a function in storybook

### DISCUSSION THREAD

**Discussion by wadakatu** [tighten/ziggy#619] [Q&A] [answered]
Hi.
Thank you for this amazing library.

I created an app with Vue3, Laravel and ziggy-js.
Then, I introduced Storybook to create UI catalog.

However, I got an error when I made a story.
This error occurs where I used route method of ziggy-js.

`_ctx.route is not a function.`

For example...

```vue
 <Button v-if="!course" :href="route('top')" primary quick>TOP</Button>
```

Is there any way to resolve this error?

**Comment by bakerkretzmar** [comment, marked as answer]
What's the entire component/store/file that error is coming from? I'm not super familiar with storybook but you'll need to make sure the `route()` function is available in your story, either by installing the Ziggy Vue plugin or manually importing it and setting it up.

**Reply by wadakatu** [reply]
@bakerkretzmar 

Thank you for your reply.
Thanks to you, I resolved the error as follows.
I created route method and made it available in my story with `app.mixin` in preview.js.

```vue
import { app } from '@storybook/vue3';

const route = (route) => {
  return '#';
};

app.mixin({ methods: { route } });
```
